src/topology/inseparable.lean

```diff
@@ -178,12 +179,197 @@ def specialization_preorder : preorder X :=
+if `x ∈ S → y ∈ S` for all `x ⤳ y`. -/
+def stable_under_specialization (s : set X) : Prop :=
+∀ ⦃x y⦄, x ⤳ y → x ∈ s → y ∈ s
+
+/-- A subset `S` of a topological space is stable under specialization
+if `x ∈ S → y ∈ S` for all `y ⤳ x`. -/
+def stable_under_generalization (s : set X) : Prop :=
+∀ ⦃x y⦄, y ⤳ x → x ∈ s → y ∈ s
+
+example {s : set X} : stable_under_specialization s ↔ is_lower_set s := iff.rfl
+example {s : set X} : stable_under_generalization s ↔ is_upper_set s := iff.rfl
+
+lemma is_closed.stable_under_specialization {s : set X} (hs : is_closed s) :
+  stable_under_specialization s :=
+λ x y e, e.mem_closed hs
+
+lemma is_open.stable_under_generalization {s : set X} (hs : is_open s) :
+  stable_under_generalization s :=
+λ x y e, e.mem_open hs
+
+@[simp]
+lemma stable_under_specialization_compl_iff {s : set X} :
+  stable_under_specialization sᶜ ↔ stable_under_generalization s :=
+is_lower_set_compl
+
+@[simp]
+lemma stable_under_generalization_compl_iff {s : set X} :
+  stable_under_generalization sᶜ ↔ stable_under_specialization s :=
+is_upper_set_compl
+
+alias stable_under_specialization_compl_iff ↔ _ stable_under_generalization.compl
+alias stable_under_generalization_compl_iff ↔ _ stable_under_specialization.compl
+
+lemma stable_under_specialization_sUnion (S : set (set X))
+  (H : ∀ s ∈ S, stable_under_specialization s) : stable_under_specialization (⋃₀ S) :=
+is_lower_set_sUnion H
+
+lemma stable_under_specialization_sInter (S : set (set X))
+  (H : ∀ s ∈ S, stable_under_specialization s) : stable_under_specialization (⋂₀ S) :=
+is_lower_set_sInter H
+
+lemma stable_under_generalization_sUnion (S : set (set X))
+  (H : ∀ s ∈ S, stable_under_generalization s) : stable_under_generalization (⋃₀ S) :=
+is_upper_set_sUnion H
+
+lemma stable_under_generalization_sInter (S : set (set X))
+  (H : ∀ s ∈ S, stable_under_generalization s) : stable_under_generalization (⋂₀ S) :=
+is_upper_set_sInter H
+
+lemma stable_under_specialization_Union {ι : Sort*} (S : ι → set X)
+  (H : ∀ i, stable_under_specialization (S i)) : stable_under_specialization (⋃ i, S i) :=
+is_lower_set_Union H
+
+lemma stable_under_specialization_Inter {ι : Sort*} (S : ι → set X)
+  (H : ∀ i, stable_under_specialization (S i)) : stable_under_specialization (⋂ i, S i) :=
+is_lower_set_Inter H
+
+lemma stable_under_generalization_Union {ι : Sort*} (S : ι → set X)
+  (H : ∀ i, stable_under_generalization (S i)) : stable_under_generalization (⋃ i, S i) :=
+is_upper_set_Union H
+
+lemma stable_under_generalization_Inter {ι : Sort*} (S : ι → set X)
+  (H : ∀ i, stable_under_generalization (S i)) : stable_under_generalization (⋂ i, S i) :=
+is_upper_set_Inter H
+
+lemma stable_under_specialization_iff_eq_Union {s : set X} :
+  stable_under_specialization s ↔ s = ⋃ x ∈ s, closure {x} :=
+show is_lower_set s ↔ _, by simp only [closure_singleton_eq_Iic, ← lower_closure_eq,
+  ← lower_set.supr_Iic, lower_set.coe_supr₂, lower_set.coe_Iic, eq_comm]
```

Comment:
Could you modify this the same way as we did to `lower_closure_eq`? Also the proof shouldn't reference `lower_set.coe_supr₂, lower_set.coe_Iic`, but rather `coe_lower_set`, as mentioned in my previous comment.